deps/rabbit/src/rabbit_queue_type.erl

```diff
@@ -765,3 +771,25 @@ known_queue_type_names() ->
     {QueueTypes, _} = lists:unzip(Registered),
     QTypeBins = lists:map(fun(X) -> atom_to_binary(X) end, QueueTypes),
     ?KNOWN_QUEUE_TYPES ++ QTypeBins.
+
+-spec check_queue_limits(amqqueue:amqqueue()) ->
+          ok |
+          {protocol_error, Type :: atom(), Reason :: string(), Args :: term()}.
+check_queue_limits(Q) ->
+    maybe
+        %% Prepare for more checks
+        ok ?= check_vhost_queue_limit(Q)
+    end.
+
+check_vhost_queue_limit(Q) ->
+    #resource{name = QueueName} = amqqueue:get_name(Q),
+    VHost = amqqueue:get_vhost(Q),
+    case rabbit_vhost_limit:is_over_queue_limit(VHost) of
+        false->
```

Comment:
Sorry but can a whitespace be added before the `->` here?

Reply:
ah, yes of course!

Reply:
Done.